### PR TITLE
Support kafka settings from secrets with load_from_secret field

### PR DESCRIPTION
**Pull request by gememma**

### CODE REVIEW

File: mirrord/operator/src/crd/kafka.rs [hunk 1]
```diff
@@ -23,6 +24,11 @@ pub struct MirrordKafkaClientConfigSpec {
     ///
     /// The list of all available properties can be found [here](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md).
     pub properties: Vec<MirrordKafkaClientProperty>,
+
+    /// Properties to set from a secret.
+    ///
+    /// The secret is resolved when resolving the rest of the final configuration.
```
Comment (Razz4780) [inline mirrord/operator/src/crd/kafka.rs:30]:
```suggestion
    /// Namespace and name of a `Secret` to use as another source of properties.
    ///
    /// The secret is fetched and entries in its `.data` field are resolved to properties.
    /// Properties resolved this way override properties from the parent configuration.
    /// Properties defined inline in this object override properties resolved from the secret.
    ///
    /// Example value: `default/my-secret`
```

File: changelog.d/+kafka-settings-from-secret.internal.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Add optional load_from_secret field to Kafka client config spec to allow setting properties from a Secret.
```
Comment (Razz4780) [inline changelog.d/+kafka-settings-from-secret.internal.md:1]:
```suggestion
Add optional `loadFromSecret` field to Kafka client config spec to allow setting properties from a Secret.
```
We rename fields to camelCase, I think it's better to use the name from resource definition (changelog is for the users)

File: mirrord/operator/src/crd/kafka.rs [hunk 3]
```diff
@@ -23,6 +24,11 @@ pub struct MirrordKafkaClientConfigSpec {
...
+    pub load_from_secret: Option<Secret>,
```
Comment (Razz4780) [inline mirrord/operator/src/crd/kafka.rs:31]:
```suggestion
    pub load_from_secret: Option<String>,
```
We don't want to include the whole `Secret` object, we want to reference an exisiting `Secret`

Comment (Razz4780) [inline mirrord/operator/src/crd/kafka.rs:31]: You can also add a `printcolumn` entry for this field (makes `kubectl get ...` output nicer)

**Review by Razz4780 — CHANGES_REQUESTED** [review]
You can a

File: mirrord/operator/src/crd/kafka.rs (hunk 3 above)
Comment (gememma) [reply mirrord/operator/src/crd/kafka.rs:31]: ohh i see 👀 ty

**Review by Razz4780 — APPROVED** [review]

**Pull request merged**